src/main/java/com/exadel/sandbox/role/service/impl/RoleServiceImpl.java

```diff
@@ -0,0 +1,89 @@
+package com.exadel.sandbox.role.service.impl;
+
+import com.exadel.sandbox.permission.dto.PerBaseDto;
+import com.exadel.sandbox.permission.dto.PerResponseDto;
+import com.exadel.sandbox.permission.entity.Permission;
+import com.exadel.sandbox.role.dto.RoleBaseDto;
+import com.exadel.sandbox.role.dto.RoleResponseDto;
+import com.exadel.sandbox.role.entity.Role;
+import com.exadel.sandbox.role.repository.RoleRepository;
+import com.exadel.sandbox.role.service.Service;
+import lombok.RequiredArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Component
+@RequiredArgsConstructor
+@Transactional
+public class RoleServiceImpl implements Service<RoleBaseDto, RoleResponseDto> {
+    private final RoleRepository roleRepository;
+    private final ModelMapper mapper;
+
+    @Override
+    public List<RoleResponseDto> getAll() {
+        List<Role> roles = roleRepository.findAll();
+        List<RoleResponseDto> roleResponseDtos = new ArrayList<>();
+        for (Role role : roles) {
+            roleResponseDtos.add(mapAllF(role));
+        }
+        return roleResponseDtos;
+    }
+
+    @Override
+    public RoleResponseDto getById(Integer id) {
+        Optional<Role> byId = roleRepository.findById(id);
+        return mapAllF(byId.orElseThrow(() -> new RuntimeException("Role not found")));
+    }
+
+    @Override
+    public RoleResponseDto create(RoleBaseDto roleBaseDto) {
+        Role role = mapper.map(roleBaseDto, Role.class);
+        List<Permission> permissions = new ArrayList<>();
+        if (roleBaseDto.getPerBaseDtos() != null) {
+            for (PerBaseDto perBaseDto : roleBaseDto.getPerBaseDtos()) {
+                permissions.add(mapper.map(perBaseDto, Permission.class));
+            }
+            role.setPermissions(permissions);
+        }
+        return mapAllF(role);
+    }
+
+    @Override
+    public void deleteById(Integer id) {
+        roleRepository.deleteById(id);
+    }
+
+    @Override
+    public RoleResponseDto update(Integer id, RoleBaseDto roleBaseDto) {
```

Comment:
For update, we should have updateDTO but not baseDTO, please check other places too